server/src/main/java/org/opensearch/index/IndexSettings.java

```diff
@@ -691,6 +691,14 @@ public static IndexMergePolicy fromString(String text) {
         Property.Dynamic
     );
 
+    public static final Setting<Integer> INDEX_CONCURRENT_SEGMENT_SEARCH_MAX_SLICE_COUNT = Setting.intSetting(
+        "index.search.concurrent.max_slice_count",
```

Comment:
should this be inline with the existing concurrent segment search setting `index.search.concurrent_segment_search.enabled`?

Something like - `index.search.concurrent_segment_search.max_slice_count` ?

Reply:
Also, would be good to add a validation that this setting can be applied only if concurrent segment search is enabled.

Reply:
Let's keep it matching the cluster setting: https://github.com/opensearch-project/OpenSearch/blob/c5c6024da6d53c145189ad898515ae4874fe7fce/server/src/main/java/org/opensearch/search/SearchService.java#L264

Also I don't think we should be validating the setting can only be applied if concurrent search is enabled, the settings are independent and should be able to be modified independently. This limitation does not exist on the existing cluster setting either.

Reply:
The cluster level setting is here 

https://github.com/opensearch-project/OpenSearch/blob/e8d3e695112c56a70b363ed5601f3b0382632bbf/server/src/main/java/org/opensearch/search/SearchService.java#L259 

I think its best to keep it consistent with that rather than have slightly different names for similar cluster/index level setting. 

Again, the corresponding cluster setting has no validation, so i was keeping it consistent with that. It anyway doesn't make sense to change slice count setting if you have concurrent search disabled.

Reply:
why is there a disparity in the setting names? 
it should either be `search.concurrent.setting_name` or `search.concurrent_segment_search.setting_name`.
It should all be consistent IMO.

Reply:
For the scope of this PR I think consistency with the cluster setting is what matters most. If you think making the change to the existing setting is required let's open a separate issue to discuss that, but for now we should be consistent with what already exists.

Reply:
i mean ideally, we should have not introduced a new variety of setting here which is the cluster setting that is not in-line with the concurrent search enabled setting.
@Gankris96 can you make sure that the settings are consistent before we make the document changes for it?(https://github.com/opensearch-project/documentation-website/issues/8068)

Reply:
What do you mean by consistent? 
Re-name all the settings to be 
`search.concurrent_segment_search.<setting_name>`  and `index.search.concurrent_segment_search.<setting_name>` ? 

And then update the documentation as well?

Reply:
+1 please open a separate issue to discuss renaming existing settings. This PR is introducing a new index setting and it should match the naming of the existing cluster setting that provides the same functionality.